datafusion/execution/src/disk_manager.rs

```diff
@@ -71,11 +72,11 @@ impl DiskManagerConfig {
 /// while processing dataset larger than available memory.
 #[derive(Debug)]
 pub struct DiskManager {
-    /// TempDirs to put temporary files in.
+    /// Dirs to put temporary files in.
     ///
-    /// If `Some(vec![])` a new OS specified temporary directory will be created
+    /// If `Some(vec![])` temporary files will be created in the directories.
     /// If `None` an error will be returned (configured not to spill)
-    local_dirs: Mutex<Option<Vec<TempDir>>>,
+    local_dirs: Mutex<Option<Vec<PathBuf>>>,
```

Comment:
Am I correct in understanding that the implication here is that the directories in `local_dirs` will not be removed (though all the contents will be removed, when the `NamedTempFile` are dropped)?

Reply:
This solution uses directories specified, or returned by `std::env::temp_dir()`, which is usually `/tmp` on Linux.
So, these directories are not deleted when the `NamedTempFile` are dropped.